### PR TITLE
refactor app

### LINKED ISSUE #70: refactor our demo app for completely verification

### PR DESCRIPTION
**Pull request by peacepassion**
refactor the demo app, and fix two bugs.   

close #70 

cc @JackCho

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,24 @@ In your module's `build.gradle`
...
+There is a app module sub project. And there is an Demo page in the app demonstrating how to apply patch apk. Please ref the usage.
+
+## Development
+#### Process
+There are two gradle tasks provided in the app/build.gradle, `:app:runHost`, `:app:preparePatch`, which can accelerate development.
+
+* `./gradlew runHost`, launch the host app
+* `./gradlew preparePatch`, build and push the patch apk to the device
+* apply the patch apk in the Demo page
+
```
Comment (peacepassion) [inline README.md]: Following this process, would be easy.

File: amigo-lib/src/main/java/me/ele/amigo/Amigo.java [hunk 2]
```diff
@@ -125,6 +125,12 @@ private void runPatchApk() throws LoadPatchApkException {
             setDexElements(amigoClassLoader);
             setNativeLibraryDirectories(amigoClassLoader);
 
+            // issue: different class loader has different static object
+            // https://coderanch.com/t/385932/java/java/ClassLoader-Static-variables
+            amigoClassLoader.loadClass("me.ele.amigo.AmigoDirs")
+                    .getDeclaredMethod("init", Context.class)
+                    .invoke(null, this);
```
Comment (peacepassion) [inline amigo-lib/src/main/java/me/ele/amigo/Amigo.java]: bug1

File: amigo-lib/src/main/java/me/ele/amigo/Amigo.java [hunk 3]
```diff
@@ -465,6 +471,13 @@ public static boolean hasWorked() {
         return classLoader != null && classLoader.getClass().getName().equals(AmigoClassLoader.class.getName());
     }
 
+    public static int patchVersion(Context ctx) {
```
Comment (peacepassion) [inline amigo-lib/src/main/java/me/ele/amigo/Amigo.java]: Used for client check if apply new patch apk.

File: amigo-lib/src/main/java/me/ele/amigo/compat/RCompat.java [hunk 4]
```diff
@@ -64,10 +66,10 @@ public static int getIdentifier(Context context, int id) {
     }
 
     private static void initClasses(Context context) {
-        if (classes[0] == null) {
+        if (classes.size() == 0) {
             for (int i = 0; i < classNames.length; i++) {
                 try {
-                    classes[i] = Class.forName(context.getPackageName() + "." + classNames[i]);
+                    classes.add(Class.forName(context.getPackageName() + "." + classNames[i]));
```
Comment (peacepassion) [inline amigo-lib/src/main/java/me/ele/amigo/compat/RCompat.java]: bug2

File: app/build.gradle [hunk 5]
```diff
@@ -29,9 +30,18 @@ android {
...
+            signingConfig signingConfigs.debug
             minifyEnabled false
+            zipAlignEnabled true
+        }
+
+        patch {
+            debuggable true
+            signingConfig signingConfigs.debug
+            minifyEnabled false
+            zipAlignEnabled true
```
Comment (peacepassion) [inline app/build.gradle]: use this two buildTypes for seperating host and patch

File: app/src/main/java/me/ele/amigo/app/HomeActivity.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        Log.e(TAG, "onCreate");
+        Log.e(TAG, "getApplication from HomeActivity-->" + getApplication());
+
+        infoView.setText(A.getDes());
+    }
+
+    @OnClick(R.id.goto_demo)
+    public void gotoDemo(View v) {
+        startActivity(new Intent(this, DemoActivity.class));
+    }
```
Comment (peacepassion) [inline app/src/main/java/me/ele/amigo/app/HomeActivity.java]: demo entry

File: app/src/main/java/me/ele/amigo/app/HomeActivity.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    @OnClick(R.id.goto_dev)
+    public void gotoDev(View v) {
+        startActivity(new Intent(this, DevActivity.class));
+    }
```
Comment (peacepassion) [inline app/src/main/java/me/ele/amigo/app/HomeActivity.java]: development entry

File: README.md [hunk 8]
```diff
@@ -63,15 +80,7 @@ There are two ways to make hotfix work.
...
 	```
 
-
-### clear the previously installed patch-apk manually
-
-```java
-Amigo.clear(context);
-```
-**note**：The old apk would be cleared automatically when the latest version updated.
-
```
Comment (JackCho) [inline README.md]: why delete this

File: amigo-lib/src/main/java/me/ele/amigo/compat/RCompat.java (hunk 4 above)
Comment (JackCho) [inline amigo-lib/src/main/java/me/ele/amigo/compat/RCompat.java]: cool

File: app/build.gradle [hunk 9]
```diff
@@ -1,12 +1,13 @@
...
 apply plugin: 'me.ele.amigo'
+apply plugin: 'com.neenbedankt.android-apt'
 
 android {
     compileSdkVersion 23
     buildToolsVersion "23.0.3"
 
     defaultConfig {
-        applicationId "me.ele.demo"
+        applicationId "me.ele.amigo.app"
```
Comment (JackCho) [inline app/build.gradle]: don't rename to `me.ele.amigo.**`, because all classes in it will be kept

File: README.md (hunk 8 above)
Comment (peacepassion) [inline README.md]: this API has been deleted.

**Pull request merged**